tests/src/test/scala/com/nvidia/spark/rapids/BroadcastHashJoinSuite.scala

```diff
@@ -71,4 +71,31 @@ class BroadcastHashJoinSuite extends SparkQueryCompareTestSuite {
       }
     })
   }
+
+  test("replace AST GpuBroadCastHashJoin with GpuBroadcastNestedLoopJoin") {
+    withGpuSparkSession(spark => {
+      longsDf(spark).createOrReplaceTempView("t")
+      longsDf(spark).createOrReplaceTempView("u")
+
+      val plan1 = spark.sql(s"SELECT /*+ BROADCAST(t) */ * FROM t JOIN u ON t.longs <= u.longs")
+      val plan2 = spark.sql(s"SELECT /*+ BROADCAST(u) */ * FROM t JOIN u ON t.longs > u.longs")
+
+      // execute the plan so that the final adaptive plan is available when AQE is on
+      plan1.collect()
+
+      val finalPlan1 = findOperator(plan1.queryExecution.executedPlan,
+        _.isInstanceOf[GpuBroadcastNestedLoopJoinExecBase])
```

Comment:
This test should be integrated into the new integration tests using `assert_cpu_and_gpu_are_equal_collect_with_capture`